### PR TITLE
[Feature] UseStmt support catalog.database

### PR DESCRIPTION
**Pull request by stephen-shelby**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

'use xxx' from mysql client will be handle by COM_INIT_DB protocol in fe, it won't generate UseStmt.
'use xxx' from jdbc will be handle by COM_QUERY, we need adapt UseStmt to 'use catalog.db'.
this pr port UseStmt from old parser and analyzer to new parser and analyzer.

### CODE REVIEW

**Comment by stephen-shelby** [comment]
cc @HangyuanLiu

**Comment by stephen-shelby** [comment]
run starrocks_fe_unittest

**Comment by stephen-shelby** [comment]
run starrocks_clang-format

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java [hunk 1]
```diff
@@ -1633,8 +1633,15 @@ public ParseNode visitShowDatabasesStatement(StarRocksParser.ShowDatabasesStatem
...
-        Identifier identifier = (Identifier) visit(context.identifier());
-        return new UseStmt(identifier.getValue());
+        QualifiedName qualifiedName = getQualifiedName(context.qualifiedName());
+        List<String> parts = qualifiedName.getParts();
+        if (parts.size() == 1) {
+            return new UseStmt(null, parts.get(0));
+        } else if (parts.size() == 2) {
+            return new UseStmt(parts.get(0), parts.get(1));
+        } else {
+            throw new ParsingException("error catalog.database");
```
Comment (HangyuanLiu) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java:1643]:
```suggestion
                        throw new StarRocksPlannerException("error qualifiedName in UseStmt",
                    ErrorType.INTERNAL_ERROR);
```

**Comment by HangyuanLiu** [comment]
Missing privilege check

**Review by DorianZheng — COMMENTED** [review]
LGTM

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java (hunk 1 above)
Comment (stephen-shelby) [reply fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java]: done

**Comment by stephen-shelby** [comment]
> Missing privilege check

We have checked the permission in GlobalStateMgr.changeCatalogDb. because the queries from mysql client will be an string. it won't be parsed. so we need call the same interface when process 'use xxx' from mysql client or jdbc.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 18 / 21 (85.71%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/UseStmt.java|9|12|75.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|7|7|100.00%|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|1|1|100.00%|

File: fe/fe-core/src/test/java/com/starrocks/analysis/UseStmtTest.java [hunk 2]
```diff
@@ -17,45 +17,78 @@
 
 package com.starrocks.analysis;
 
-import com.starrocks.common.AnalysisException;
-import com.starrocks.common.UserException;
+import com.starrocks.catalog.Database;
```
Comment (imay) [inline fe/fe-core/src/test/java/com/starrocks/analysis/UseStmtTest.java:20]: seems all the content has been changed. Better change the license header to 
```
// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
```

Comment (stephen-shelby) [reply fe/fe-core/src/test/java/com/starrocks/analysis/UseStmtTest.java:5]: done

**Review by gengjun-git — APPROVED** [review]

**Comment by stephen-shelby** [comment]
@Mergifyio backport branch-2.3

**Comment by stephen-shelby** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 18 / 21 (85.71%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/UseStmt.java|9|12|75.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|7|7|100.00%|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|1|1|100.00%|

**Review by HangyuanLiu — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 18 / 21 (85.71%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/UseStmt.java|9|12|75.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|7|7|100.00%|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|1|1|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 18 / 21 (85.71%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/UseStmt.java|9|12|75.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|7|7|100.00%|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|1|1|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 18 / 21 (85.71%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/UseStmt.java|9|12|75.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|7|7|100.00%|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|1|1|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 18 / 21 (85.71%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/UseStmt.java|9|12|75.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|7|7|100.00%|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|1|1|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 18 / 21 (85.71%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/UseStmt.java|9|12|75.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|7|7|100.00%|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|1|1|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 18 / 21 (85.71%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/UseStmt.java|9|12|75.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|7|7|100.00%|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|1|1|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 18 / 21 (85.71%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/UseStmt.java|9|12|75.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|7|7|100.00%|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|1|1|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 18 / 21 (85.71%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/UseStmt.java|9|12|75.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|7|7|100.00%|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|1|1|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 18 / 21 (85.71%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/UseStmt.java|9|12|75.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|7|7|100.00%|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|1|1|100.00%|

**Pull request merged**